src/target/riscv/riscv-013.c

```diff
@@ -934,34 +918,39 @@ static int dm013_select_target(struct target *target)
 	return dm013_select_hart(target, info->index);
 }
 
-static int execute_abstract_command(struct target *target, uint32_t command,
-		uint32_t *cmderr)
+static int batch_run_timeout(struct target *target, struct riscv_batch *batch);
+
+static int execute_abstract_command_in_batch(struct target *target,
+		struct riscv_batch *batch, uint32_t command, uint32_t *cmderr)
 {
-	assert(cmderr);
-	*cmderr = CMDERR_NONE;
-	if (debug_level >= LOG_LVL_DEBUG) {
-		switch (get_field(command, DM_COMMAND_CMDTYPE)) {
-			case 0:
-				LOG_DEBUG_REG(target, AC_ACCESS_REGISTER, command);
-				break;
-			default:
-				LOG_TARGET_DEBUG(target, "command=0x%x", command);
-				break;
-		}
-	}
+	riscv_batch_add_dm_write(batch, DM_COMMAND, command, /* read_back */ true,
+			RISCV_DELAY_BASE);
+	riscv_batch_add_dm_read(batch, DM_ABSTRACTCS, RISCV_DELAY_ABSTRACT_COMMAND);
```

Comment:
The write to DM_COMMAND triggers the abstract command execution, so it looks to me that the delays should be the other way around:

```suggestion
	riscv_batch_add_dm_write(batch, DM_COMMAND, command, /* read_back */ true,
			RISCV_DELAY_ABSTRACT_COMMAND);
	riscv_batch_add_dm_read(batch, DM_ABSTRACTCS, RISCV_DELAY_BASE);
```

Reply:
Thanks for the catch! Addressed.